README.md
<div align="center" markdown="1">

# [`app-desktop-useless-cpp-gui`][url-repo]

[![License][badge-license]][url-license]
[![Status][badge-status-abandoned]][url-repo]
[![Version][badge-version]][url-version]
[![Release][badge-workflow-release]][url-release]
[![Coverage][badge-sonar-coverage]][url-sonar]
[![Lines of Code][badge-sonar-ncloc]][url-sonar]
[![Conventional Commits][badge-conventionalcommits]][url-conventionalcommits]

[![Get as an Appimage][badge-appimage]][url-release-latest]

Desktop GUI

</div>

## About The Project

In 2020 I've decided to make an app. For some reason, 
I've decided to use C++ and Qt5 and to make a GUI first. 
Later when I started implementing logic, I realized C++ 
is not suitable for that particular app, so I abandoned it.

### Features

- Basic C++, Qt5
- Packaged as an AppImage

### Usage

```bash
curl https://api.github.com/repos/shishifubing/app-desktop-useless-cpp-gui/releases/latest \
  | jq -r .assets[].browser_download_url                                                   \
  | xargs wget
chmod +x app-desktop-useless-cpp-gui.AppImage
./app-desktop-useless-cpp-gui.AppImage
```

### Demo

> **Note**
>
> WebM is broken on Edge, try another browser

[url-demo]

<!-- relative links -->

<!-- project links -->

[url-repo]: https://github.com/shishifubing/app-desktop-useless-cpp-gui
[url-license]: https://github.com/shishifubing/app-desktop-useless-cpp-gui/blob/main/LICENSE
[url-release-latest]: https://github.com/shishifubing/app-desktop-useless-cpp-gui/releases/latest
[url-workflow-build]: https://github.com/shishifubing/app-desktop-useless-cpp-gui/actions/workflows/build.yml?branch=main
[url-sonar]: https://sonarcloud.io/dashboard?id=shishifubing_app-desktop-useless-cpp-gui
[url-release]: https://github.com/shishifubing/app-desktop-useless-cpp-gui/actions/workflows/release.yml
[url-version]: https://github.com/shishifubing/app-desktop-useless-cpp-gui/releases/latest

<!-- external links -->

[url-conventionalcommits]: https://conventionalcommits.org
[url-demo]: https://user-images.githubusercontent.com/97828377/218940781-ffbede8c-d3a1-497c-aabf-484bcea08b12.webm

<!-- project badge links -->

[badge-workflow-release]: https://img.shields.io/github/actions/workflow/status/shishifubing/app-desktop-useless-cpp-gui/release.yml?branch=main&label=release&logo=github
[badge-version]: https://img.shields.io/github/v/release/shishifubing/app-desktop-useless-cpp-gui?label=version
[badge-license]: https://img.shields.io/github/license/shishifubing/app-desktop-useless-cpp-gui.svg
[badge-status-abandoned]: https://img.shields.io/badge/status-abandoned-red


<!-- other badge links -->

[badge-conventionalcommits]: https://img.shields.io/badge/conventional--commits-1.0.0-%23FE5196?logo=conventionalcommits&logoColor=white
[badge-appimage]: https://raw.githubusercontent.com/AppImage/docs.appimage.org/master/source/_static/img/download-appimage-banner.svg

<!-- sonar badge links -->

[badge-sonar-vulnerabilities]: https://sonarcloud.io/api/project_badges/measure?project=shishifubing_app-desktop-useless-cpp-gui&metric=vulnerabilities
[badge-sonar-sqale_index]: https://sonarcloud.io/api/project_badges/measure?project=shishifubing_app-desktop-useless-cpp-gui&metric=sqale_index
[badge-sonar-security_rating]: https://sonarcloud.io/api/project_badges/measure?project=shishifubing_app-desktop-useless-cpp-gui&metric=security_rating
[badge-sonar-reliability_rating]: https://sonarcloud.io/api/project_badges/measure?project=shishifubing_app-desktop-useless-cpp-gui&metric=reliability_rating
[badge-sonar-alert_status]: https://sonarcloud.io/api/project_badges/measure?project=shishifubing_app-desktop-useless-cpp-gui&metric=alert_status
[badge-sonar-sqale_rating]: https://sonarcloud.io/api/project_badges/measure?project=shishifubing_app-desktop-useless-cpp-gui&metric=sqale_rating
[badge-sonar-ncloc]: https://sonarcloud.io/api/project_badges/measure?project=shishifubing_app-desktop-useless-cpp-gui&metric=ncloc
[badge-sonar-duplicated_lines_density]: https://sonarcloud.io/api/project_badges/measure?project=shishifubing_app-desktop-useless-cpp-gui&metric=duplicated_lines_density
[badge-sonar-coverage]: https://sonarcloud.io/api/project_badges/measure?project=shishifubing_app-desktop-useless-cpp-gui&metric=coverage
[badge-sonar-code_smells]: https://sonarcloud.io/api/project_badges/measure?project=shishifubing_app-desktop-useless-cpp-gui&metric=code_smells
[badge-sonar-bugs]: https://sonarcloud.io/api/project_badges/measure?project=shishifubing_app-desktop-useless-cpp-gui&metric=bugs


CHANGELOG.md
# CHANGELOG

## v0.3.52 - 2023-03-12

### Documentation

- **readme**: switch github badges to shields.io by @tiandaoburen in [#60](https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/60)

**Full Changelog**: https://github.com/shishifubing/app-desktop-useless-cpp-gui/compare/v0.3.44...v0.3.52

## v0.3.44 - 2023-03-11

<!-- Release notes generated using configuration in .github/release.yml at v0.3.44 -->
### What's Changed

#### Documentation

- Add v0.3.38 to CHANGELOG.md [skip ci] by @shishifubing-bot in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/44
- Add Usage section by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/45

#### Other Changes

- Remove issue labeler workflow by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/46
- Use a reusable workflow in the release workflow by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/47
- Add 'needs' to the release workflow, modify exclude section in the config by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/49
- Replace hashes with branches by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/51

**Full Changelog**: https://github.com/shishifubing/app-desktop-useless-cpp-gui/compare/v0.3.38...v0.3.44

## v0.3.38 - 2023-03-11

<!-- Release notes generated using configuration in .github/release.yml at v0.3.38 -->
### What's Changed

#### Documentation

- Add badges to README.md by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/43

### New Contributors

- @shishifubing-bot made their first contribution in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/39
- @dependabot made their first contribution in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/41

**Full Changelog**: https://github.com/shishifubing/app-desktop-useless-cpp-gui/compare/v0.3.33...v0.3.38

## v0.3.33 - 2023-03-11

<!-- Release notes generated using configuration in .github/release.yml at v0.3.33 -->
### What's Changed

#### Bug Fixes

- Remove outline from selected elements by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/1

#### Documentation

- Add screencast by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/2
- Add 'features' section by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/4
- Add appimage banner by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/5
- Center the readme header by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/7
- Add sonarqube badges by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/10
- Add a line about sonarqube to README.md by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/11
- Move sonarqube badges lower by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/12
- Add 'About The Project' section by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/13
- Remove 'Badges' section by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/14
- Change headings by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/15
- Replace the build badge by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/20
- Add badges by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/22
- Add sonar badges by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/25
- Rearrange badges by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/33
- Change wording by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/34

#### Other Changes

- Add sonar-project.properties by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/8
- Remove local GitVersion config by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/27
- Create CODEOWNERS by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/32
- Add CHANGELOG rule to CODEOWNERS by @tiandaoburen in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/38

### New Contributors

- @tiandaoburen made their first contribution in https://github.com/shishifubing/app-desktop-useless-cpp-gui/pull/1

**Full Changelog**: https://github.com/shishifubing/app-desktop-useless-cpp-gui/compare/v0.1.0...v0.3.33


GUI/mainwindow.cpp
#include "mainwindow.h"
#include "ui_mainwindow.h"
#include <menubutton.h>
#include <systembutton.h>
#include <QDebug>
#include <QGuiApplication>
#include <QString>

MainWindow::MainWindow(QWidget *parent) :
    QMainWindow(parent),
    ui(new Ui::MainWindow)
{
    ui->setupUi(this);
    setMainWindow();
    setMenuFrame();
    setContentFrame();
    fillMenuPanel();
}

MainWindow::~MainWindow()
{
    delete ui;
}

void MainWindow::setMainWindow()
{
    QWidget::setWindowFlags(Qt::FramelessWindowHint);
    ui->Main_Window->setFixedSize(1024,600);
    QHBoxLayout *mainWindow_Layout = new QHBoxLayout;
    ui->Main_Window->setLayout(mainWindow_Layout);
    mainWindow_Layout->setMargin(0);
    mainWindow_Layout->setSpacing(0);
    ui->Main_Window->setMaximumSize(QGuiApplication::primaryScreen()->availableSize());
    QFrame *menuFrame = new QFrame, *contentFrame = new QFrame;
    this->menuFrame=menuFrame; this->contentFrame=contentFrame;
    menuFrame->setFixedWidth(224);
    menuFrame->setStyleSheet("background-color: rgb(60, 60, 60);");
    contentFrame->setStyleSheet("border-top:1px solid rgb(217, 0, 255);"
                                 "border-left:1px solid rgb(217, 0, 255);");
    contentFrame->setFrameShape(QFrame::Box);
    contentFrame->setFrameShadow(QFrame::Plain);
    mainWindow_Layout->addWidget(menuFrame);
    mainWindow_Layout->addWidget(contentFrame);
}

void MainWindow::setMenuFrame()
{
    QVBoxLayout *frame_Layout = new QVBoxLayout;
    this->menuFrame->setLayout(frame_Layout);
    frame_Layout->setContentsMargins(10,10,10,10);
    QStackedWidget *menuPanel = new QStackedWidget;
    this->menuPanel=menuPanel;
    frame_Layout->addWidget(menuPanel);
    menuPanel->setStyleSheet("background-color: rgb(50, 50, 50); "
                             "border-radius: 50px");
    menuPanel->setMaximumHeight(
               QGuiApplication::primaryScreen()->availableSize().height()*2/3);
    QWidget *menuPanelPage = new QWidget;
    menuPanel->addWidget(menuPanelPage);
    QVBoxLayout *menuPanelPage_Layout = new QVBoxLayout;
    menuPanelPage_Layout->setAlignment(Qt::AlignTop);
    menuPanelPage_Layout->setSpacing(35);
    menuPanelPage->setLayout(menuPanelPage_Layout);
    this->menuPanelPage_Layout=menuPanelPage_Layout;
}

void MainWindow::setContentFrame()
{
    QVBoxLayout *frameLayout = new QVBoxLayout;
    this->contentFrame->setLayout(frameLayout);
    QHBoxLayout *systemButtonsLayout = new QHBoxLayout;
    frameLayout->addLayout(systemButtonsLayout);
    systemButtonsLayout->setAlignment(Qt::AlignRight);
    systemButtonsLayout->setSpacing(0);
    SystemButton *minimize = new SystemButton(this, 0),
                 *maximize = new SystemButton(this, 1),
                    *close = new SystemButton(this, 2);
    systemButtonsLayout->addWidget(minimize);
    systemButtonsLayout->addWidget(maximize);
    systemButtonsLayout->addWidget(close);
    QStackedWidget *contentPanel = new QStackedWidget;
    this->contentPanel=contentPanel;
    contentPanel->setStyleSheet("border: none;");
    frameLayout->addWidget(contentPanel);
}

void MainWindow::fillMenuPanel()
{
    this->amountOfMenuPages++;
    if (this->menuPanel->currentIndex()) {
        addNewMenuButton("", 4, ":/Img/Img/home.png");
    }
    else {
        addNewMenuButton("", 2, ":/Img/Img/home.png");
    }
    addNewMenuButton("", 6, ":/Img/Img/add.png");
    addNewMenuButton("", 6, ":/Img/Img/add.png");
    addNewMenuButton("", 6, ":/Img/Img/add.png");
    addNewMenuButton("", 6, ":/Img/Img/add.png");
    addNewMenuButton("", 6, ":/Img/Img/add.png");
    addNewMenuButton("", 6, ":/Img/Img/add.png");
    addNewMenuButton("", 3, ":/Img/Img/add.png");
}

void MainWindow::addNewMenuButton(QString text, int type,
                               QString iconPath)
{
    MenuButton *newMenuButton = new MenuButton(this, text, type, iconPath);
    QHBoxLayout *newMenuButtonLayout = new QHBoxLayout;
    newMenuButtonLayout->setAlignment(Qt::AlignHCenter);
    if (type==2 || type==3 || type==4) {
        if (type==3) {
            this->menuPanelPage_Layout->addStretch();
        }
        MenuButton *newMenuArrowLeft =
            new MenuButton(this, "", -51, ":/Img/Img/leftArrow.png");
        MenuButton *newMenuArrowRight =
            new MenuButton(this, "", -52, "");
        if (type==2 || type==4) {
            this->menuArrowTop=newMenuArrowRight;
        }
        else if (type==3) {
            this->menuArrowBot=newMenuArrowRight;
        }
        newMenuButtonLayout->addWidget(newMenuArrowLeft);
        newMenuButtonLayout->addWidget(newMenuButton);
        newMenuButtonLayout->addWidget(newMenuArrowRight);
        this->addButton=newMenuButton;
        this->menuPanelPage_Layout->addLayout(newMenuButtonLayout);
        if (type!=3) {
            this->menuPanelPage_Layout->addStretch();
            newMenuButton->indexOfContentPage=0;
            if (this->amountOfContentPages==0) {
                addContentPage(newMenuButton);
            }
        }
        if (!this->menuPanel->currentIndex()) {
            newMenuArrowLeft->setIcon(QPixmap(""));
        }
    }
    else {
        newMenuButtonLayout->addWidget(newMenuButton);
        this->menuPanelPage_Layout->addLayout(newMenuButtonLayout);
    }
}

void MainWindow::addButtonWasClicked(MenuButton* button)
{
    if (button->type==3) {
        QWidget *menuPanelPage = new QWidget;
        this->menuPanel->addWidget(menuPanelPage);
        QVBoxLayout *menuPanelPage_Layout = new QVBoxLayout;
        menuPanelPage_Layout->setSpacing(35);
        menuPanelPage->setLayout(menuPanelPage_Layout);
        this->menuPanelPage_Layout=menuPanelPage_Layout;
        this->menuPanel->setCurrentIndex(this->menuPanel->currentIndex()+1);
        this->menuArrowTop->setIcon(QPixmap(":/Img/Img/rightArrow.png"));
        this->menuArrowBot->setIcon(QPixmap(":/Img/Img/rightArrow.png"));
        this->addButton->setIcon(QPixmap(""));
        this->addButton->disconnect(this->addButton,
                                    SIGNAL(addButtonWasClicked(MenuButton*)),
                                    this, SLOT(addButtonWasClicked(MenuButton*)));
        fillMenuPanel();
    }
    else if (button->type==6) {
        button->type=0;
        button->connectMenuButton(this);
        addContentPage(button);
    }
}

void MainWindow::addContentPage(MenuButton *button)
{
    button->indexOfContentPage=this->amountOfContentPages++;
    QWidget *contentPanelPage = new QWidget;
    QGridLayout *contentPanelPage_Layout = new QGridLayout;
    contentPanelPage_Layout->setAlignment(Qt::AlignCenter);
    contentPanelPage->setLayout(contentPanelPage_Layout);
    this->contentPanel->addWidget(contentPanelPage);
    int x = 0, y = 0;
    for (int i=0; i<=button->indexOfContentPage; i++) {
        QPushButton *count = new QPushButton;
        QString number, label;
        if (!button->indexOfContentPage) {
            count->setStyleSheet("background-color: rgb(60,0,100); color: white;"
                                 "border-radius: 75px");
            label="ContentPage_Home";
            count->setText(label);
            count->setFont(QFont("Century Gothic", 10, QFont::Normal));
            count->setFixedSize(150,150);
        }
        else if (button->indexOfContentPage<16){
            count->setStyleSheet("background-color: rgb(60,60,60); color: white;"
                             "border-radius: 75px");
            number.setNum(button->indexOfContentPage);
            label="ContentPage_"+number;
            count->setText(label);
            count->setFont(QFont("Century Gothic", 10, QFont::Normal));
            count->setFixedSize(150,150);
        }
        else {
            count->setStyleSheet("background-color: rgb(60,0,100); color: white;"
                             "border-radius: 37px");
            number.setNum(button->indexOfContentPage);
            label="ContentPage_"+number;
            count->setText(label);
            count->setFont(QFont("Century Gothic", 5, QFont::Normal));
            count->setFixedSize(74,74);
        }
        if (i>=button->indexOfContentPage && i) {
            break;
        }
        contentPanelPage_Layout->addWidget(count, x, y++);
        if (y==5 && button->indexOfContentPage<16) {
            y=0;
            x++;
        }
        else if (y==7) {
            y=0;
            x++;
        }
    }
    QString number, label;number.setNum(button->indexOfContentPage);
    if (button->indexOfContentPage) {
        button->setFixedSize(184,40);
        button->setIcon(QPixmap(""));
        label="MenuButton_"+number;
        button->setText(label);
    }
}

void MainWindow::emitMenuButtonWasClicked()
{
    emit menuButtonWasClicked();
}

void MainWindow::minimizeApp()
{
    this->showMinimized();
    emit systemButtonWasClicked();
}

void MainWindow::maximizeApp()
{
    if (isMaximized()) {
        this->showNormal();
        emit systemButtonWasClicked();
    }
    else {
        this->showMaximized();

    }
}

void MainWindow::closeApp()
{
    emit systemButtonWasClicked();
    this->close();
}

void MainWindow::setContentPanelIndex(int indexOfContentPage)
{
    this->contentPanel->setCurrentIndex(indexOfContentPage);
}

void MainWindow::setMenuPanelIndex(int index)
{
    if (index==-51 && this->menuPanel->currentIndex()) {
        this->menuPanel->setCurrentIndex(this->menuPanel->currentIndex()-1);
    }
    else if(index==-52 &&
            this->menuPanel->currentIndex()+1<this->amountOfMenuPages) {
            setMenuPanelIndex(this->menuPanel->currentIndex()+1);
    }
    else if (index>=0){
        this->menuPanel->setCurrentIndex(index);
    }
}




GUI/ui_mainwindow.h
/********************************************************************************
** Form generated from reading UI file 'mainwindow.ui'
**
** Created by: Qt User Interface Compiler version 5.12.2
**
** WARNING! All changes made in this file will be lost when recompiling UI file!
********************************************************************************/

#ifndef UI_MAINWINDOW_H
#define UI_MAINWINDOW_H

#include <QtCore/QVariant>
#include <QtWidgets/QApplication>
#include <QtWidgets/QMainWindow>
#include <QtWidgets/QWidget>

QT_BEGIN_NAMESPACE

class Ui_MainWindow
{
public:
    QWidget *Main_Window;

    void setupUi(QMainWindow *MainWindow)
    {
        if (MainWindow->objectName().isEmpty())
            MainWindow->setObjectName(QString::fromUtf8("MainWindow"));
        MainWindow->resize(1024, 600);
        MainWindow->setWindowOpacity(1.000000000000000);
        MainWindow->setStyleSheet(QString::fromUtf8(""));
        Main_Window = new QWidget(MainWindow);
        Main_Window->setObjectName(QString::fromUtf8("Main_Window"));
        Main_Window->setStyleSheet(QString::fromUtf8("background-color: rgb(212, 207, 201);"));
        MainWindow->setCentralWidget(Main_Window);

        retranslateUi(MainWindow);

        QMetaObject::connectSlotsByName(MainWindow);
    } // setupUi

    void retranslateUi(QMainWindow *MainWindow)
    {
        MainWindow->setWindowTitle(QApplication::translate("MainWindow", "MainWindow", nullptr));
    } // retranslateUi

};

namespace Ui {
    class MainWindow: public Ui_MainWindow {};
} // namespace Ui

QT_END_NAMESPACE

#endif // UI_MAINWINDOW_H


GUI/menubutton.cpp
#include "menubutton.h"
#include <QPixmap>

/*
States:
        0: Normal, 1: Hover, 2: Clicked;
Types:
        0: no icon menu button, 1: custom icon menu button,
        2: home menu button, 3: add menu button,
        4: home menu button not clicked, -51: left arrow, -52: right arrow,
        6: add button without arrows
*/
MenuButton::MenuButton(QWidget *parent, QString text, int type, QString iconPath) : QPushButton(parent)
{
    setMenuButton(text, type, iconPath);
    connectMenuButton(parent);
}

void MenuButton::setMenuButton(QString text, int type, QString iconPath)
{
    setMouseTracking(true);
    setAttribute(Qt::WA_Hover);
    this->type=type;
    this->setFlat(true);
    this->setFont(QFont("Century Gothic", 13, QFont::Normal));
    if (type==2) {
        this->setButtonStyleSheet(2);
        this->clicked=true;
    }
    else this->setButtonStyleSheet();
    if (text==QString("")) {
        this->setFixedSize(40,40);
    }
    else {
        this->setText(text);
        this->setFixedSize(184,40);
    }
    if (type) {
        this->setIcon(QPixmap(iconPath));
        if (type==-51 || type==-52) {
            this->setIconSize(QSize(20,20));
            this->setFixedSize(25,25);
        }
        else {
            this->setIconSize(QSize(40,40));
        }
    }
}

void MenuButton::connectMenuButton(QWidget *parent)
{
    if (this->type==3 || this->type==6) {
        connect(this, SIGNAL(addButtonWasClicked(MenuButton*)),
                parent, SLOT(addButtonWasClicked(MenuButton*)));
    }
    else if (this->type==-51 || this->type==-52) {
        connect(this, SIGNAL(setMenuPanelIndex(int)),
                parent, SLOT(setMenuPanelIndex(int)));
    }
    else {
        connect(this, SIGNAL(emitMenuButtonWasClicked()),
        parent, SLOT(emitMenuButtonWasClicked()));
        connect(parent, SIGNAL(menuButtonWasClicked()),
                this, SLOT(menuButtonWasClicked()));

        connect(this, SIGNAL(setContentPanelIndex(int)),
                parent, SLOT(setContentPanelIndex(int)));
    }
}

void MenuButton::menuButtonWasClicked()
{
    if (this->clicked) {
        setButtonStyleSheet();
        this->clicked=false;
    }
}


void MenuButton::setButtonStyleSheet(int state)
{
    switch(state) {
    case 0:
        if (this->type==-51 || this->type==-52)
            this->setStyleSheet("background-color: rgb(60, 60, 60); border-radius: 10px;"
                                "outline: none;"
                                "border: 1px solid rgb(60, 0, 100); color: white;");
        else
            this->setStyleSheet("background-color: rgb(60, 60, 60); border-radius: 20px;"
                                "outline: none;"
                                "border: 1px solid rgb(60, 0, 100); color: white;");
        break;
    case 1:
        if (this->type==-51 || this->type==-52)
            this->setStyleSheet("background-color: rgb(60, 0, 100); border-radius: 10px;"
                                "outline: none;"
                                "border: 1px solid rgb(60, 0, 100); color: white;");
        else
            this->setStyleSheet("background-color: rgb(60, 0, 100); border-radius: 20px;"
                                "outline: none;"
                                "border: 1px solid rgb(60, 0, 100); color: white;");
        break;
    case 2:
        this->setStyleSheet("background-color: rgb(60, 0, 100); border-radius: 20px;"
                            "border: 1px solid rgb(60, 0, 100); color: white;"
                            "outline: none;"
                            "border-left: 4px solid rgb(217, 0, 255);"
                            "border-right: 4px solid rgb(217, 0, 255);");
        break;
    case 3:
        if (this->type==-51 || this->type==-52)
            this->setStyleSheet("background-color: rgb(60, 0, 100); border-radius: 10px;"
                                "outline: none;"
                                "border: 2px solid rgb(217, 0, 255); color: white;");
        else
            this->setStyleSheet("background-color: rgb(60, 0, 100); border-radius: 20px;"
                                "outline: none;"
                                "border: 2px solid rgb(217, 0, 255); color: white;");
        break;
    }
}


bool MenuButton::event(QEvent *event)
{
    switch(event->type())
    {
    case QEvent::HoverEnter:
        if (!this->clicked)
            setButtonStyleSheet(1);
        return true;
    case QEvent::HoverLeave:
        if (!this->clicked)
            setButtonStyleSheet();
        return true;
    case QEvent::MouseButtonPress:
        setButtonStyleSheet(3);
        return true;
    case QEvent::MouseButtonRelease:
        if (this->type==3 || this->type==6) {
            emit addButtonWasClicked(this);
            setButtonStyleSheet();
        }
        else if (this->type==-51 || this->type==-52) {
            setButtonStyleSheet();
            emit setMenuPanelIndex(this->type);
        }
        else {
            emit setContentPanelIndex(this->indexOfContentPage);
            emit emitMenuButtonWasClicked();
            this->clicked=true;
            setButtonStyleSheet(2);
        }
        return true;
    default:
        break;
    }
    return QWidget::event(event);
}



